tests/sparseml/pytorch/recipe_template/test_main.py

```diff
@@ -0,0 +1,135 @@
+)
+
+
+@pytest.mark.parametrize(
+    "pruning, quantization, kwargs",
+    [
+        ("true", True, {}),
+        ("true", True, {"global_sparsity": True}),
+        ("true", False, {}),
+        ("false", True, {}),
+        ("false", False, {}),
+    ],
+)
+def test_generic_recipe_creation(
+    pruning: str, quantization: bool, kwargs, model: Optional[Module]
+):
+    actual = _build_recipe(pruning=pruning, quantization=quantization, **kwargs)
+    assert actual
+    manager = ScheduledModifierManager.from_yaml(file_path=actual)
+    manager.apply(module=model)
+
+
+@pytest.mark.parametrize(
+    "pruning, quantization, kwargs",
+    [
+        ("true", True, {"lr_func": "cosine"}),
+        ("true", False, {"lr_func": "cyclic_linear"}),
+        ("false", True, {}),
+        ("false", False, {}),
+    ],
+)
+def test_recipe_creation_with_a_specific_model(
+    pruning: str, quantization: bool, kwargs, model: Optional[Module]
+):
+    actual = _build_recipe(
+        pruning=pruning, quantization=quantization, model=model, **kwargs
+    )
+    assert actual
+    manager = ScheduledModifierManager.from_yaml(file_path=actual)
+    manager.apply(module=model)
+
+
+def test_recipe_can_be_updated():
+    actual = _build_recipe(pruning="true", quantization=False)
+    manager = ScheduledModifierManager.from_yaml(
+        file_path=actual,
+        recipe_variables=dict(
+            start_epoch=100,
+            end_epoch=1000,
+        ),
+    )
+    recipe_from_manager = str(manager)
+    assert "start_epoch: 100" in recipe_from_manager
+    assert "end_epoch: 1000" in recipe_from_manager
+
+
+@pytest.mark.parametrize(
+    "pruning_algo, expected",
+    [
+        ("true", "!MagnitudePruningModifier"),
+        ("acdc", "!ACDCPruningModifier"),
+        ("mfac", "!MFACPruningModifier"),
+        ("movement", "!MovementPruningModifier"),
+        ("constant", "!ConstantPruningModifier"),
+    ],
+)
+def test_pruning_modifiers_match_pruning_algo(pruning_algo: str, expected: str):
+    actual_recipe = _build_recipe(pruning=pruning_algo)
+    manager = ScheduledModifierManager.from_yaml(file_path=actual_recipe)
+    manager_recipe = str(manager)
+    assert expected in manager_recipe
+
+
+@min_torch_version
+def test_obs_modifier():
+    test_pruning_modifiers_match_pruning_algo(
+        pruning_algo="obs", expected="!OBSPruningModifier"
+    )
+
+
+@pytest.mark.parametrize(
+    "pruning, quantization, quant_expected",
+    [
+        ("true", True, True),
+        ("true", False, False),
+        ("false", True, True),
+        ("false", False, False),
+    ],
+)
+def test_one_shot_applies_sparsification(pruning, quantization, quant_expected, model):
+    actual = _build_recipe(pruning=pruning, quantization=quantization, model=model)
+    manager = ScheduledModifierManager.from_yaml(file_path=actual)
+    manager.apply(module=model)
+    model_is_quantized = hasattr(model, "qconfig") and model.qconfig is not None
+
```

Comment:
need to assert that pruning is applied when applicable - can iterate through the prunable parameters in the module and assert that the `tensor_sparsity` is as expected.

use this util function: https://github.com/neuralmagic/sparseml/blob/0f5f3a8a39a4881b46f68a69a050d26eaae19adc/src/sparseml/pytorch/utils/helpers.py#L550